WhateverGreen/kern_igfx_memory.cpp

```diff
@@ -0,0 +1,199 @@
+//
+//  kern_igfx_memory.cpp
+//  WhateverGreen
+//
+//  Created by FireWolf on 8/30/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#include "kern_igfx.hpp"
+#include <Headers/kern_util.hpp>
+#include <Headers/kern_disasm.hpp>
+
+void IGFX::DVMTCalcFix::init() {
+	// We only need to patch the framebuffer driver
+	requiresPatchingGraphics = false;
+	requiresPatchingFramebuffer = true;
+}
+
+void IGFX::DVMTCalcFix::deinit() {}
+
+void IGFX::DVMTCalcFix::processKernel(KernelPatcher &patcher, DeviceInfo *info) {
+	// Guard: Disable the patch if it is not available on the current Intel platforms
+	if (!available) {
+		SYSLOG("igfx", "DVMT: This fix is not available on the current platform and has been disabled.");
+		return;
+	}
+		
+	// Enable the fix if designated boot argument or device property is found
+	enabled = checkKernelArgument("-igfxdvmt");
+	if (!enabled)
+		enabled = info->videoBuiltin->getProperty("enable-dvmt-calc-fix") != nullptr;
+	
+	// Read the DVMT preallocated memory set in BIOS from the GMCH Graphics Control field at 0x50 (PCI0,2,0)
```

Comment:
I do not believe this code should execute when `enabled` is false.

Reply:
```
// Enable the fix if designated boot argument or device property is found
enabled = checkKernelArgument("-igfxdvmt");
if (!enabled)
	enabled = info->videoBuiltin->getProperty("enable-dvmt-calc-fix") != nullptr;
```

I believe that the code enables the patch if either the boot argument exists or the device property is defined.
I remember that this is the usual way to enable a patch, as I was corrected by someone else in a PR last year.
Are there any issues?

Reply:
I meant that the code after these checks has no real benefit to execute when `enable` is false.

Reply:
Ah, my bad. Sorry about that. I was thinking of always reading the DVMT value and adding it to the registry.

Fixed.